lib/src/git_backend.rs

```diff
@@ -420,6 +423,30 @@ fn gix_open_opts_from_settings(settings: &UserSettings) -> gix::open::Options {
         .open_path_as_is(true)
 }
 
+/// Reads the `jj:trees` header from the commit.
+///
+/// Returns Some(Some(tree_id)) on success, Some(None) if the header wasn't
+/// present, and None on error.
+fn root_tree_from_header(git_commit: &CommitRef) -> Option<Option<MergedTreeId>> {
```

Comment:
nit: (at your option) this could be `Result<Option, &'static str>` or something for extra type safety.

Reply:
I avoided returning a specific error because I don't think we need more detail for troubleshooting, but I agree about the type safety (I wasn't happy with `Option<Option<...>>`). Oh, to signal an error but without additional detail, I suppose `Result<T, ()>` is reasonable. I'll switch to that.